pkg/cli/internal/commands/deploy.go

```diff
@@ -107,6 +108,178 @@ func init() {
+  kmcp deploy package --deployment-name my-server --manager uvx --args mcp-server-git --uv-env                                             # Use UV and write managed tools and installables to /tmp directories`,
+	Args: cobra.NoArgs,
+	RunE: runPackageDeploy,
+}
+
+func init() {
+	// package subcommand flags
+	packageDeployCmd.Flags().StringVar(&packageName, "deployment-name", "", "Name for the deployment (required)")
+	packageDeployCmd.Flags().StringVar(&packageManager, "manager", "", "Package manager to use (npx or uvx) (required)")
+	packageDeployCmd.Flags().StringSliceVar(&packageSecrets, "secrets", []string{}, "List of Kubernetes secret names to mount")
+	// UV-specific flags
+	packageDeployCmd.Flags().BoolVar(&packageUvEnv, "uv-env", false, "Add UV-specific environment variables for writing managed tools and installables to /tmp directories. Useful when not using a custom image to avoid permission issues.")
+	// Add common deployment flags
+	packageDeployCmd.Flags().StringSliceVar(&deployArgs, "args", []string{}, "Arguments to pass to the package manager (e.g., package names) (required)")
+	packageDeployCmd.Flags().StringSliceVar(&deployEnv, "env", []string{}, "Environment variables (KEY=VALUE)")
+	packageDeployCmd.Flags().BoolVar(&deployDryRun, "dry-run", false, "Generate manifest without applying to cluster")
+	packageDeployCmd.Flags().StringVarP(&deployNamespace, "namespace", "n", "", "Kubernetes namespace")
+	packageDeployCmd.Flags().StringVar(&deployImage, "image", "", "Docker image to deploy (overrides default)")
+	packageDeployCmd.Flags().StringVar(&deployTransport, "transport", "", "Transport type (stdio, http)")
+	packageDeployCmd.Flags().IntVar(&deployPort, "port", 0, "Container port (default: 3000)")
+	packageDeployCmd.Flags().BoolVar(&deployNoInspector, "no-inspector", false, "Do not start the MCP inspector after deployment")
+	packageDeployCmd.Flags().StringVarP(&deployOutput, "output", "o", "", "Output file for the generated YAML")
+
+	// Mark required flags
+	_ = packageDeployCmd.MarkFlagRequired("deployment-name")
+	_ = packageDeployCmd.MarkFlagRequired("manager")
+	_ = packageDeployCmd.MarkFlagRequired("args")
+}
+
+func runPackageDeploy(_ *cobra.Command, args []string) error {
+	// Validate package manager
+	if packageManager != "npx" && packageManager != "uvx" {
+		return fmt.Errorf("unsupported package manager: %s (must be 'npx' or 'uvx')", packageManager)
+	}
+
+	// Validate args
+	if len(deployArgs) == 0 {
+		return fmt.Errorf("args are required (e.g., --args package-name)")
+	}
+
+	// Parse environment variables
+	envMap := parseEnvVars(deployEnv)
+	// Add UV-specific environment variables to create directories
+	// in tmp directory to avoid permission issues when not using a custom image
+	if packageManager == "uvx" && packageUvEnv {
+		envMap["UV_INSTALL_DIR"] = "/tmp/uv"
+		envMap["UV_CACHE_DIR"] = "/tmp/uv/cache"
+		envMap["UV_TOOL_DIR"] = "/tmp/uv/tools"
+		envMap["UV_TOOL_BIN_DIR"] = "/tmp/uv/tools/bin"
+	}
```

Comment:
this is necessary when using the default `ghcr.io/astral-sh/uv:debian` image unless a custom image is provided which has permissions to create the cache and tools directory

Reply:
removed this after relaxing the security context.